src/metatracer.rs

```diff
@@ -30,20 +30,60 @@
 // DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 
-use std::sync::atomic::{AtomicU32, Ordering};
+use std::{
+    cell::RefCell,
+    sync::{atomic::{AtomicU32, Ordering}, Arc, RwLock},
+    thread,
+};
+use hwtracer::{Trace, Tracer, ThreadTracer, HWTracerError, TracerState};
+#[cfg(hwtracer_perf_pt)]
+use hwtracer::backends::{perf_pt::PerfPTTracer};
+
+thread_local! {
+    // Each thread has its own `ThreadTracer`, initialised lazily on first use.
+    pub static THREAD_TRACER: RefCell<Option<Box<ThreadTracer>>> = RefCell::new(None);
```

Comment:
Could this be `Option<RefCell<Box<ThreadTracer>>>`? That way we wouldn't even have to allocate a `RefCell` in many cases.

Reply:
Hmm, or is this whole thing completely lazily initialised anyway? That's what `LocalKey::with` suggests...

Reply:
If we put the `Option` on the outside, then we can't mutate it later. That's what the `RefCell` is giving us: inner mutability.

This is lazily initialised, but it's nothing to do with the local key, that's just the Rust thread local API. We lazily change a `None` to a `Some` the first time a new thread wants to trace.

Reply:
Yeah, I was an idiot on the `Option` thing.

But my reading of the thread local API is that this variable will be lazily initialised, so we don't even need to bother with the inner `Option`?